File: ch6/clip1/boj2998/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String binary = sc.next();
        if(binary.length() % 3 == 1)
            binary = "00" + binary;
        else if(binary.length() % 3 == 2)
            binary = "0" + binary;

        change(binary);
    }
    public static int o2d(String binary) {
        final String[] octal = {
            "000", "001", "010",
            "011", "100", "101",
            "110", "111"
        };
        for (int i = 0; i < 8; i++) {
            if(binary.equals(octal[i])) return i;
        }
        return -1;
    }
    public static void change (String binary) {
        int length = binary.length();
        if(length == 0) return;

        change(binary.substring(0, length - 3));
        int d = o2d(binary.substring(length - 3));
        System.out.print(d);
    }
}

File: ch6/clip1/boj2998/main.py
def o2d(binary):
    octal = [
        "000", "001", "010",
        "011", "100", "101",
        "110", "111"
    ]
    for i in range(8):
        if binary == octal[i]:
            return i
    return -1

def change(binary):
    length = len(binary)
    if length == 0:
        return

    change(binary[:-3])
    d = o2d(binary[-3:])
    print(d, end="")

binary = input()
if len(binary) % 3 == 1:
    binary = "00" + binary
elif len(binary) % 3 == 2:
    binary = "0" + binary

change(binary)
